File: src/226.invert_binary_tree.kt
fun main() {
    println(invertTree(
        TreeNode(4).apply {
            left = TreeNode(4).apply {
                left = TreeNode(1)
                right = TreeNode(3)
            }
            right = TreeNode(2).apply {
                left = TreeNode(6)
                right = TreeNode(9)
            }
        }
    ))

    println(invertTree(
        TreeNode(2).apply {
            left = TreeNode(1)
            right = TreeNode(3)
        }
    ))
}

fun invertTree(root: TreeNode?): TreeNode? {
    if (root == null || root.left == null && root.right == null) {
        return root
    }

    val temp = root.left
    root.left = invertTree(root.right)
    root.right = invertTree(temp)

    return root
}

File: lib/226.invert_binary_tree.dart
import 'other/tree_node.dart';

void main() {
  print(invertTree(TreeNode(4, TreeNode(2, TreeNode(1), TreeNode(3)),
      TreeNode(7, TreeNode(6), TreeNode(9)))));
  print(invertTree(TreeNode(2, TreeNode(1), TreeNode(3))));
}

TreeNode? invertTree(TreeNode? root) {
  if (root == null || root.left == null && root.right == null) {
    return root;
  }

  final temp = root.left;
  root.left = invertTree(root.right);
  root.right = invertTree(temp);

  return root;
}
